src/device/DeviceBase.cpp

```diff
@@ -1049,6 +1049,94 @@ std::vector<CameraBoardSocket> DeviceBase::getConnectedCameras() {
     return pimpl->rpcClient->call("getConnectedCameras").as<std::vector<CameraBoardSocket>>();
 }
 
+std::vector<StereoPair> DeviceBase::getAvailableStereoPairs() {
+    std::vector<dai::StereoPair> stereoPairs;
+    std::function<std::vector<dai::CameraBoardSocket>(dai::CameraBoardSocket, const std::unordered_map<dai::CameraBoardSocket, dai::CameraInfo>&, std::vector<dai::CameraBoardSocket>&)> visitLinks;
+    visitLinks = [&visitLinks](dai::CameraBoardSocket camId, const std::unordered_map<dai::CameraBoardSocket, dai::CameraInfo>& camData, std::vector<dai::CameraBoardSocket>& visited) -> std::vector<dai::CameraBoardSocket> {
```

Comment:
Perhaps the following would be simpler (if it goes through)
```suggestion
    auto visitLinks = [&visitLinks](dai::CameraBoardSocket camId, const std::unordered_map<dai::CameraBoardSocket, dai::CameraInfo>& camData, std::vector<dai::CameraBoardSocket>& visited) -> std::vector<dai::CameraBoardSocket> {
```

Reply:
Unfortunately can't use auto here..
```
error: use of ‘visitLinks’ before deduction of ‘auto’
```

Reply:
Oh right, recursion